pecos/xmc/xlinear/model.py

```diff
@@ -437,37 +438,56 @@ def predict(
             selected_outputs_csr (csr_matrix, optional): instance label matrix to predict from with
                 shape (instances, labels). Interested labels to predict are denoted only by indices
                 with a nonzero value.
+            batch_size (int, optional): the number of instances to predict in a batch.
             kwargs:
                 beam_size (int, optional): override the beam size specified in the model.
                     Default None to disable overriding
                 only_topk (int, optional): override the only topk specified in the model
                     Default None to disable overriding
                 post_processor (str, optional):  override the post_processor specified in the model
                     Default None to disable overriding
-                threads (int, optional): the number of threads to use for training.
+                threads (int, optional): the number of threads to use for prediction.
                     Defaults to -1 to use all
 
         Returns:
             Y_pred (csr_matrix): If a selected output matrix is provided, the prediction for the
                 indicated output is returned. Otherwise, a prediction matrix for some topk
                 scores is returned.
         """
-        if selected_outputs_csr is None:
-            if pred_params is None:
-                Y_pred = self.model.predict(X, pred_params=None, **kwargs)
-            elif isinstance(pred_params, self.PredParams):
-                Y_pred = self.model.predict(X, pred_params=pred_params.hlm_args, **kwargs)
+        if batch_size is None:
+            if selected_outputs_csr is None:
+                if pred_params is None:
```

Comment:
Can we change this (and line466) to `Y_pred = self.model.predict(X, pred_params=None if pred_params is None else pred_params.hlm_args, **kwargs)`

Reply:
Thanks for the suggestion. I think this original style was aiming to catch and raise the error for invalid types of pred_params, but the suggested style indeed has better readability. I will change it to the suggested style and add a check for the instance type ahead.